### DISCUSSION TITLE
ObservableArray with JS array concat

### DISCUSSION THREAD

**Discussion by lgibso34** [mobxjs/mobx#3042] [Q&A] [answered]
https://www.mobxjs.com/refguide/array.html#array-limitations-in-mobx-4-and-below
the above link states that only MOBX 4 and below have the `concat` issue but I am still experiencing this in MOBX 6.

Issue:
```javascript
const test: ObservableArray = [item1, item2, item3]

const newArray = [].concat(test); // returns [[item1, item2, item3]] instead of [item1, item2, item3]
```

So, it puts the full array on the inside instead of spreading it. The solution is either `.slice()` the test array or spread (...) it out.

I am just wondering if this should be fixed after MOBX 4 or if the documentation for this `concat` issue is just a bit out of date or something.

Running: 
```json
    "mobx": "^6.2.0",
    "mobx-react": "^7.1.0",
```

**Comment by urugator** [comment, marked as answer]
The official documentation is here (mobxjs.com isn't ours):
https://mobx.js.org/README.html

The behavior depends on proxy configuration and availability - without proxy the behavior matches V4, with proxy the behavior matches V5:
https://codesandbox.io/s/arrayconcat-xigy2?file=/index.js

**Reply by lgibso34** [reply]
Thank you. Great codesandbox for the explanation!

EDIT: Here is what i found from the official documentation: https://mobx.js.org/configuration.html#limitations-without-proxy-support
